content/en/devlog/tweasel-update-4/index.md

```diff
@@ -0,0 +1,70 @@
+    * [How should we archive the external links that we use as sources for our documentation?](https://github.com/tweaselORG/meta/issues/36) This one is particularly tricky. We ideally want to use an established and trusted service like the [Wayback Machine](https://web.archive.org/) but that has issues with lots of the pages we tried (sites blocking the archiving resulting in broken snapshots, dynamic content not being captured properly). The only other public alternative is archive.today, which is heavily gated behind reCAPTCHA and cannot be automated. Meanwhile, the available self-hosted solutions all have their own problems, and we wouldn't be able to host them publicly anyway due to copyright issues. In the end, we decided to create our own solution, keeping that as simple as possible:
+        * We store the original links in the `reasoning` field of the adapters, and have a file watcher that tries to archive them using the [Save Page Now API](https://archive.org/details/spn-2-public-api-page-docs-2023-01-22) of archive.org. Errors are recorded in a local file so we can retry using an exponential backoff strategy. Lorenz has mostly [implemented](https://github.com/tweaselORG/TrackHAR/pull/54) this already.
+        
+          The archived links are stored in a CSV file in the TrackHAR repo and have redirects set up in a subfolder on `trackers.tweasel.org`.
+        * If automatic archiving fails, we have to manually create a snapshot on a public archive, likely using archive.today and add that to the CSV.
+        * We will manually take screenshots and PDFs of the pages, and upload them to a Nextcloud folder or similar. These can unfortunately not be hosted publicly.
+        * We'll create a [linting script](https://github.com/tweaselORG/TrackHAR/issues/33) that warns us on commit if some of the links have not been archived.
+    * What do we even want to document, how in-depth do we want to go? Do we always need to reference official documentation if available? We decided that we will only document things related to the properties in our adapters to avoid scope creep, even though there are a lot of (disturbingly) interesting admissions on tracker websites. It is fine to use "obvious" as a reasoning if the property name or value is self-explanatory. We do not need to seek out official documentation for every property, but we should update our documentation if we find it later. We also decided that we won't document properties that are listed in official documentation but that we have not observed in our requests, as we only rely on the data we find.
```

Comment:
```suggestion
    * What do we even want to document, how in-depth do we want to go? Do we always need to reference official documentation if available?  
We decided that we will only document things related to the properties in our adapters to avoid scope creep, even though there are a lot of (disturbingly) interesting admissions on tracker websites. It is fine to use "obvious" as a reasoning if the property name or value is self-explanatory. We do not need to seek out official documentation for every property, but we should update our documentation if we find it later. We also decided that we won't document properties that are listed in official documentation but that we have not observed in our requests, as we only rely on the data we find.
```